odl/trafos/non_uniform_fourier.py

```diff
@@ -9,123 +9,242 @@
 """Discretized non-uniform Fourier transform on L^p spaces."""
 
 from __future__ import division
+
 import numpy as np
 
-from odl import DiscreteLp, cn
+from odl.discr import DiscreteLp
 from odl.operator import Operator
-from odl.trafos import PYNFFT_AVAILABLE
+from odl.space import cn
+from odl.space.base_tensors import TensorSpace
+from odl.trafos.backends.pynfft_bindings import (
+    PYNFFT_AVAILABLE, normalize_samples)
+from odl.util import complex_dtype
+
 if PYNFFT_AVAILABLE:
-    from pynfft.nfft import NFFT
+    import pynfft
 
 
 class NonUniformFourierTransformBase(Operator):
-    """Non uniform Fast Fourier Transform.
-    """
-    def __init__(
-        self, space, samples, domain, range, skip_normalization=False):
+
+    """Base class for non-uniform Fourier Transform and its adjoint."""
```

Comment:
I think we have to specify fast fourier transform, since this involves some degree of approximation.

Reply:
You're right.